modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -482,4 +513,61 @@ private void setOperationIsAsync() {
             additionalProperties.put(OPERATION_IS_ASYNC, operationIsAsync);
         }
     }
+
+    private void setIsFramework() {
+        if (aspnetCoreVersion.getOptValue().startsWith("3.")) {
+            // default, do nothing
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so changing  to use frameworkReference instead of packageReference ");
+            isFramework = true;
+            additionalProperties.put(IS_FRAMEWORK, isFramework);
+        } else {
+            if (additionalProperties.containsKey(IS_FRAMEWORK)) {
+                isFramework = convertPropertyToBooleanAndWriteBack(IS_FRAMEWORK);
+            } else {
+                additionalProperties.put(IS_FRAMEWORK, isFramework);
+            }
+        }
+
+    }
+
+    private void setUseNewtonsoft() {
+        if (aspnetCoreVersion.getOptValue().startsWith("2.")) {
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so staying on default json library.");
+            useNewtonsoft = false;
```

Comment:
I feel like there's either a typo here or on the conditional a couple lines above. Is it intentional to disallow JSON.net in aspnetcore 2.x? If so, this may be considered a breaking change and should probably be done separately from this PR to generate discussion.

Reply:
The change is not a breaking change:

This one is a little tricky - the MVC configuration in 2.x and 3.x use AddJsonOptions but mean different things: the default JSON in 2.x is implicitly Newtonsoft, while in 3.x and later it will be the Microsoft one. 

I can change it use something like useDefaultJson or jsonLibrary. I don't like the former since the default in different cases means different things. Maybe explicitly defining the jsonLibrary as Newtonsoft or Microsoft but that makes the mustache file complex - since mustache does not have conditionals based on values.

What are your thoughts there?

Reply:
My main concern is that for 2.x, when `useNewtonSoft` is false, this results in the NewtonSoft library not being referenced, but is still used on controllers and models. From a maintainability perspective (both built-in templates and custom user templates), I find that very confusing.

Reply:
For `2.x` `NewtonSoft` is used by the framework by default itself because it is part of the fw, but MS removed it from the framework starting from `3.x` and manual addition of package is needed. 
Which in this case the user really has no choice to not use NewtonSoft in `2.x`
Since we would need to use `NewtonSoft` for model compatibility reasons it does not makes sense to allow the user to set this flag because if the fw is > `3.x` we will need to add it